[TAN-2315] Fix Bug in Posting Limits

adessy opened the pull request:
The posting limits, which dictate the number of `Idea` records a user can create within a participation context, were previously encoded using two attributes in the `Phase` model:
- `posting_method` ('limited' or 'unlimited') indicating if the limit should be enforced
- `posting_limited_max` (integer) specifying the actual limit

This was a bit over-engineered since, today, all phases have a `posting_limit` of 1, and the limit is (in principle) never enforced except for native surveys.

However, these attributes were not updated when the participation method changed, leading to inconsistencies:
- Some native surveys allowed unlimited submissions
- Other participation types, like ideation, could enforce a limit of 1 idea per user

To resolve this issue and given the underuse of these attributes, we've decided to remove them and hardcode the limit (if any) at the participation method level.


# Changelog
## Fixed
- [TAN-2315] Fix bug in posting limits when changing the participation method of a phase.

adessy committed:
[TAN-2315] Fix Bug in Posting Limits

The posting limits, which dictate the number of `Idea` records a user can create within a participation context, were previously encoded using two attributes in the `Phase` model:
- `posting_method` ('limited' or 'unlimited') indicating if the limit should be enforced
- `posting_limited_max` (integer) specifying the actual limit

This was a bit over-engineered since, today, all phases have a `posting_limit` of 1, and the limit is (in principle) never enforced except for native surveys.

However, these attributes were not updated when the participation method changed, leading to inconsistencies:
- Some native surveys allowed unlimited submissions
- Other participation types, like ideation, could enforce a limit of 1 idea per user

To resolve this issue and given the underuse of these attributes, we've decided to remove them and hardcode the limit (if any) at the participation method level.

adessy committed:
Merge branch 'refs/heads/master' into TAN-2315/incorrect-posting-limits-when-changing-participation-method

# Conflicts:
#	back/db/structure.sql

jamesspeake commented on back/app/services/permissions/phase_permissions_service.rb:
```diff
@@ -151,15 +151,16 @@ def volunteering_denied_reason_for_phase
+      limit = phase.pmethod.posting_limit
+      return false unless limit
+
+      nb_posts = phase.ideas.where(author: user, publication_status: 'published').size
```
personal preference possibly, but pref `num` to `nb` when shortening number

adessy replied:
> In English, nb is not a common abbreviation for "number." It's more often used in contexts where French influence is present, as "nb" stands for "nombre" (French for "number").
> (ChatGPT, circa 2020)

jamesspeake reviewed (approved):
Much much nicer way of doing this! Only one tiny comment, but nothing blocking

adessy committed:
[TAN-2315] Fix abbreviation for "number"

Adressing comments from review.

> In English, nb is not a common abbreviation for "number." It's more often used in contexts where French influence is present, as "nb" stands for "nombre" (French for "number").
  (ChatGPT, circa 2020)
